File: web/src/core/rules/diffCalculation.ts
// Spec: docs/specs/rules/diff-calculation.md
// Mudanças aqui DEVEM começar pela spec. Esta função consolida a regra de Δ
// que antes vivia duplicada em PersonCard.tsx e AcertoPage.tsx.

import type { Row, Person } from "../types";
import { splitForPerson } from "./splitForPerson";

export function diffCalculation(rows: Row[], person: Person): number {
  const other: Person = person === "Julio" ? "Dani" : "Julio";
  const monthHasPix = rows.some((r) => r.origem === "Pix (contas)");
  let meu = 0;
  let outro = 0;
  if (monthHasPix) {
    for (const r of rows) {
      if (r.origem !== "Pix (contas)") continue;
      meu += splitForPerson(r, person);
      outro += splitForPerson(r, other);
    }
  } else {
    for (const r of rows) {
      if (r.origem === "Contas" || r.origem === "Empregados") {
        meu += splitForPerson(r, person);
        outro += splitForPerson(r, other);
      }
    }
  }
  return meu - outro;
}

File: app/lib/core/rules/diff_calculation.dart
// Spec: docs/specs/rules/diff-calculation.md
// Mudanças aqui DEVEM começar pela spec.

import '../types.dart';
import 'split_for_person.dart';

double diffCalculation(List<ExpenseRow> rows, Person person) {
  final other = person.other;
  final monthHasPix = rows.any((r) => r.origem == 'Pix (contas)');
  double meu = 0;
  double outro = 0;
  if (monthHasPix) {
    for (final r in rows) {
      if (r.origem != 'Pix (contas)') continue;
      meu += splitForPerson(r, person);
      outro += splitForPerson(r, other);
    }
  } else {
    for (final r in rows) {
      if (r.origem == 'Contas' || r.origem == 'Empregados') {
        meu += splitForPerson(r, person);
        outro += splitForPerson(r, other);
      }
    }
  }
  return meu - outro;
}
